synapse/util/file_consumer.py

```diff
@@ -80,9 +82,13 @@ def registerProducer(
         self.streaming = streaming
         self._finished_deferred = run_in_background(
             threads.deferToThreadPool,
-            self._reactor,
-            self._reactor.getThreadPool(),
-            self._writer,
+            # mypy seems to get confused with the chaining of ParamSpec from
+            # run_in_background to deferToThreadPool.
+            #
+            # For Twisted trunk, ignore arg-type; for Twisted release ignore unused-ignore.
+            self._reactor,  # type: ignore[arg-type,unused-ignore]
+            self._reactor.getThreadPool(),  # type: ignore[arg-type,unused-ignore]
+            self._writer,  # type: ignore[arg-type,unused-ignore]
```

Comment:
I don't really like this, but I don't see another option for getting this to work?

Reply:
Note that these ignores are here for Twisted release, the Twisted Trunk job ignore unused ignores.

Reply:
> I don't really like this, but I don't see another option for getting this to work?

Let's roll with this. (ParamSpec chaining has been a source of pain for me, too. I think it tends not to go well if you pass it into an overloaded function?)